plugins/command-manager/src/main/java/com/wazuh/commandmanager/index/CommandIndex.java

```diff
@@ -98,6 +101,55 @@ public CompletableFuture<RestStatus> asyncCreate(Document document) {
         return future;
     }
 
+    /**
+     * @param documents list of instances of the document model to persist in the index.
+     * @return A CompletableFuture with the RestStatus response from the operation
+     */
+    public CompletableFuture<RestStatus> asyncBulkCreate(ArrayList<Document> documents) {
+        CompletableFuture<RestStatus> future = new CompletableFuture<>();
+        ExecutorService executor = this.threadPool.executor(ThreadPool.Names.WRITE);
+
+        // Create index template if it does not exist.
+        if (!indexTemplateExists(CommandManagerPlugin.COMMAND_MANAGER_INDEX_TEMPLATE_NAME)) {
+            putIndexTemplate(CommandManagerPlugin.COMMAND_MANAGER_INDEX_TEMPLATE_NAME);
+        } else {
+            log.info(
+                    "Index template {} already exists. Skipping creation.",
+                    CommandManagerPlugin.COMMAND_MANAGER_INDEX_TEMPLATE_NAME);
+        }
+
+        BulkRequest bulkRequest = new BulkRequest();
+        for (Document document : documents) {
+            log.info("Indexing command with id [{}]", document.getId());
+            try {
+                IndexRequest request =
+                        new IndexRequest()
+                                .index(CommandManagerPlugin.COMMAND_MANAGER_INDEX_NAME)
+                                .source(
+                                        document.toXContent(
+                                                XContentFactory.jsonBuilder(),
+                                                ToXContent.EMPTY_PARAMS))
+                                .id(document.getId())
+                                .create(true);
+                bulkRequest.add(request);
+            } catch (IOException e) {
+                log.error("Error indexing command with id [{}] due to {}", document.getId(), e);
```

Comment:
This error log could be misleading as this code block prepares the bulk requests but does not execute it.